butterfree/clients/spark_client.py

```diff
@@ -58,7 +58,9 @@ def read(
         if not isinstance(options, dict):
             raise ValueError("options needs to be a dict with the setup configurations")
 
-        df_reader = self.conn.readStream if stream else self.conn.read
+        df_reader: Union[
+            DataStreamReader, DataFrameReader
+        ] = self.conn.readStream if stream else self.conn.read
```

Comment:
There is a need to declare the type of new variables? This is something the Mypy checks?

Reply:
Yes, when we have more than one possible type, an example when the annotation is not needed:
```python
x = 42
foo = 10 if x > 30 else 11  # Just one type is possible here, `int`
```